test/functional/lua/snippet_spec.lua

```diff
@@ -216,4 +216,15 @@ describe('vim.snippet', function()
     feed('foo')
     eq({ 'public function foo() {', '\t', '}' }, buf_lines(0))
   end)
+
+  it('jumps through adjacent tabstops', function ()
+    test_expand_success(
+    { 'for i=1,${1:to}${2:,step} do\n\t$3\nend' },
+    { 'for i=1,to,step do', '\t', 'end' })
+    sleep(10)
+    feed('10')
+    feed('<Tab>')
+    feed(',2')
```

Comment:
```suggestion
    poke_eventloop()
    feed('10')
    poke_eventloop()
    feed('<Tab>')
    poke_eventloop()
    feed(',2')
```

Reply:
Could you please explain the difference between `sleep` and `poke_eventloop`?

Reply:
- `sleep()` uses a hard-coded timeout.
- `poke_eventloop()` wait for the next iteration of the loop in `state_enter()`

Reply:
And... `poke_eventloop()` is needed after the `<Tab>` because input is queued, but it is not needed after `test_expand_success()` because `exec_lua()` doesn't return until the code is fully executed.

Reply:
Got it, thanks for explaining!